contact.py

```diff
@@ -6,22 +6,30 @@ def __init__(self):
         self.db = []
         self.id = 0
         # helper dictionaries:
-        
+        self.id_userid = {}
         self.id_index_dict = {}
         self.id_phone_dict = {}
-    def insert(self,client_name,name,phone):
-        
-        entry = {   'client_name':client_name,
-                    'name':name,
-                    'phone': phone,
-                    'id':  self.id }
-        self.db.append(entry)
+        self.userid_id = {}
```

Comment:
Is this for getting a contact's user id? If so, you don't need a separate index!